src/Raven.Server/Documents/Handlers/ReplicationHandler.cs

```diff
@@ -66,12 +76,65 @@ public async Task GetAllItems()
                     TimeSeriesRead = new OutgoingReplicationStatsScope(runStats),
                 };
 
-                var items = ReplicationDocumentSender.GetReplicationItems(Database, context, etag, stats, false)
-                    .Take(pageSize);
-                
+                var items = ReplicationDocumentSender.GetReplicationItems(Database, context, etag == 0 ? 0 : etag - 1, stats, false);
+                if(types.Length > 0)
+                    items = items.Where(x => types.Contains(x.Type));
+                items = items.Take(pageSize);
+
+                var debugItems = items.Select(x => x.ToDebugJson());
+
+                switch (format)
+                {
+                    case "json":
+                        await WriteJsonReplicationItems(context, debugItems);
+                        break;
+                    case "csv":
+                        await WriteCsvReplicationItems(columns, debugItems);
+                        break;
+                }
+            }
+        }
+
+        private async Task WriteCsvReplicationItems(string[] columns, IEnumerable<DynamicJsonValue> debugItems)
+        {
+            if (columns.Length == 0)
+                columns =
+                [
+                    nameof(ReplicationBatchItem.Type), nameof(ReplicationBatchItem.Etag), nameof(ReplicationBatchItem.ChangeVector),
+                    nameof(ReplicationBatchItem.LastModifiedTicks), nameof(ReplicationBatchItem.TransactionMarker), nameof(ReplicationBatchItem.Size)
+                ];
+            var encodedCsvFileName = Uri.EscapeDataString($"replication-all-items_{SystemTime.UtcNow.ToString("yyyyMMdd_HHmm", CultureInfo.InvariantCulture)}.csv");
+                        
+            HttpContext.Response.Headers.ContentDisposition = $"attachment; filename=\"{encodedCsvFileName}\"; filename*=UTF-8''{encodedCsvFileName}";
+            HttpContext.Response.Headers[Constants.Headers.ContentType] = "text/csv";
+
+            await using (var writer = new StreamWriter(HttpContext.Response.Body, Encoding.UTF8))
+            await using (var csvWriter = new CsvWriter(writer, new CsvConfiguration(CultureInfo.InvariantCulture) { Delimiter = "," }))
+            {
+                foreach (string column in columns)
+                {
+                    csvWriter.WriteConvertedField(column, typeof(string));
+                }
+                await csvWriter.NextRecordAsync();
+                foreach (var item in debugItems)
+                {
+                    foreach (string column in columns)
+                    {
+                        var value = item[column];
+                        csvWriter.WriteConvertedField(value?.ToString(), typeof(string));
```

Comment:
if string is null this is fine?

Reply:
No, I changed it to `WriteField` that doesn't ignore 'null'